### PR TITLE
retry if status code is 409

### PR DESCRIPTION
**Pull request by devidask27**
## ISSUE(S):

https://platform9.atlassian.net/browse/FT-369
## SUMMARY

Ideally if API fails in first attempt then it should retry while node authorisation but it was not retrying for API status code 409 

## ISSUE TYPE

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that may cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## IMPACTED FEATURES/COMPONENTS:

prep-node

#### Manual

Screenshot 2022-04-21 at 11 53 00 AM

### CODE REVIEW

File: pkg/util/helper.go [hunk 1]
```diff
@@ -5,11 +5,12 @@ import (
 	"context"
 	"crypto/x509"
 	"fmt"
-	"go.uber.org/zap"
 	"net/http"
 	"net/url"
 	"os"
 	"regexp"
+
```
Comment (nikhilbandi) [inline pkg/util/helper.go:12]: Why this extra line?

File: pkg/util/helper.go [hunk 2]
```diff
@@ -58,15 +59,15 @@ func RetryPolicyOn404(ctx context.Context, resp *http.Response, err error) (bool
 	// 429 Too Many Requests is recoverable. Sometimes the server puts
 	// a Retry-After response header to indicate when the server is
 	// available to start processing request from client.
-	if resp.StatusCode == http.StatusTooManyRequests {
+	if resp.StatusCode == http.StatusTooManyRequests || resp.StatusCode == http.StatusConflict {
```
Comment (nikhilbandi) [inline pkg/util/helper.go:62]: Can you please add a comment for the 409 / StatusConflict error regarding why should it be retried?

File: pkg/util/helper.go [hunk 3]
```diff
@@ -58,15 +59,15 @@ func RetryPolicyOn404(ctx context.Context, resp *http.Response, err error) (bool
...
 		return true, nil
 	}
 
 	// Check the response code. We retry on 500-range responses to allow
 	// the server time to recover, as 500's are typically not permanent
 	// errors and may relate to outages on the server side. This will catch
 	// invalid response codes as well, like 0 and 999.
-	if resp.StatusCode == 0 || resp.StatusCode == 400 || resp.StatusCode == 404 || (resp.StatusCode >= 500 && resp.StatusCode != 501) {
+	if resp.StatusCode == 0 || resp.StatusCode == http.StatusBadRequest || resp.StatusCode == http.StatusNotFound || (resp.StatusCode >= http.StatusInternalServerError && resp.StatusCode != http.StatusNotImplemented) {
```
Comment (nikhilbandi) [inline pkg/util/helper.go:70]: Should we map the status messages to the error codes in comment, just for others to understand what error codes we are handling?
e.g. StatusBadRequest = 400

**Review by nikhilbandi — CHANGES_REQUESTED** [review]
Provided minor suggestions. Please check.

File: pkg/util/helper.go (hunk 1 above)
Comment (devidask27) [reply pkg/util/helper.go:12]: Please ignore this. its because of editor.

File: pkg/util/helper.go (hunk 3 above)
Comment (devidask27) [reply pkg/util/helper.go:70]: I think no need, they are standard codes.

File: pkg/util/helper.go (hunk 2 above)
Comment (devidask27) [reply pkg/util/helper.go:64]: Done

**Review by nikhilbandi — APPROVED** [review]
Approving.

**Pull request merged**